util/containers/lru.go

```diff
@@ -37,32 +68,61 @@ func (c *LruCache[K, V]) Get(key K) (V, bool) {
 }
 
 func (c *LruCache[K, V]) Remove(key K) {
+	if c.inner == nil {
+		return
+	}
 	c.inner.Remove(key)
 }
 
+func (c *LruCache[K, V]) GetOldest() (K, V, bool) {
+	var emptyKey K
+	var emptyValue V
+	if c.inner == nil {
+		return emptyKey, emptyValue, false
+	}
+	key, value, ok := c.inner.GetOldest()
+	if !ok {
+		return emptyKey, emptyValue, false
+	}
+	castedKey, ok := key.(K)
+	if !ok {
+		panic("LRU cache has key of wrong type")
+	}
+	castedValue, ok := value.(V)
+	if !ok {
+		panic("LRU cache has value of wrong type")
+	}
+	return castedKey, castedValue, true
+}
+
 func (c *LruCache[K, V]) RemoveOldest() {
+	if c.inner == nil {
+		return
+	}
 	c.inner.RemoveOldest()
 }
 
 func (c *LruCache[K, V]) Len() int {
+	if c.inner == nil {
+		return 0
+	}
 	return c.inner.Len()
 }
 
 func (c *LruCache[K, V]) Clear() {
-	c.inner.Clear()
-}
-
-func (c *LruCache[K, V]) GetSize() int {
-	return c.inner.MaxEntries
+	if c.inner == nil {
+		return
+	}
+	c.inner.Purge()
 }
 
 func (c *LruCache[K, V]) Resize(newSize int) {
-	c.inner.MaxEntries = newSize
 	if newSize <= 0 {
-		c.Clear()
+		c.inner = nil
```

Comment:
if c.inner != nil {
 c.inner.Purge() // should trigger onEvict
}

Reply:
I've added a purge if onEvict is non-nil

Reply:
I've added a purge if onEvict is non-nil